openedx/core/djangoapps/user_api/accounts/tests/test_views.py

```diff
@@ -509,3 +510,37 @@ def test_patch_serializer_save_fails(self, serializer_save):
             error_response.data["developer_message"]
         )
         self.assertIsNone(error_response.data["user_message"])
+
+
+@unittest.skipUnless(settings.ROOT_URLCONF == 'lms.urls', 'Test only valid in lms')
+class TestAccountAPITransactions(TransactionTestCase):
+    """
+    Tests the transactional behavior of the account API
+    """
+    test_password = "test"
+
+    def setUp(self):
+        super(TestAccountAPITransactions, self).setUp()
+        self.client = APIClient()
+        self.user = UserFactory.create(password=self.test_password)
+        self.url = reverse("accounts_api", kwargs={'username': self.user.username})
+
+    @patch('openedx.core.djangoapps.user_api.accounts.serializers.AccountLegacyProfileSerializer.save')
+    def test_update_account_settings_rollback(self, user_profile_serializer_save):
+        """
+        Verify that updating preferences is transactional when a failure happens.
+        """
+        # Send a PATCH request with updates to both the user and the profile models.
```

Comment:
Good catch. I forgot to test that this one fails first (I did for the other one), and it doesn't as you say. I've tried several variations on patching but have been unable to turn this into a failing test. For some reason when I patch methods in the student Django app, they are not mocked when the code is actually reached. I'm out of time for this tonight so I'll try again tomorrow.